src/tests/ke/suite/dpc.c

```diff
@@ -0,0 +1,279 @@
+#include <xboxkrnl/xboxkrnl.h>
+#include <string.h>
+
+#include "util/output.h"
+#include "assertions/defines.h"
+
+// TODO: Add list of _LEVEL suffix support into nxdk
+#define DPC_DISPATCH_LEVEL 2 // DISPATCH_LEVEL
+
+typedef struct _dpc_test {
+    BOOLEAN inserted;
+    BOOLEAN removal;
+    BOOL dpc_called;
+    KDPC dpcObject;
+} dpc_test;
+
+static ULONG getDpcRoutineActive()
+{
+    ULONG getter;
+    __asm__ ("movl %%fs:0x58, %0"
+            : "=r"(getter));
+    return getter;
+}
+
+static void setDpcRoutineActive(ULONG setter)
+{
+    __asm__ ("movl %0, %%fs:0x58"
+            :
+            : "a"(setter));
+}
+
+static KDPC dpcObject;
+static BOOL dpc_called;
+static ULONG dpc_executing;
+static void __stdcall dpc_callback(
+	IN KDPC* Dpc,
+	IN PVOID DeferredContext,
+	IN PVOID SystemArgument1,
+	IN PVOID SystemArgument2
+)
+{
+    // DPC called variable
+    if (SystemArgument1 && (DWORD)SystemArgument1 > 0xFF) {
```

Comment:
What is this check for `> 0xFF`?

Reply:
They are tested for SystemArgument# from DPC's object structure. I assumed you didn't notice 0x#0 from KeInsertQueueDpc calls? Therefore should not be used to set a pointer's value from the callback ***if*** the callback did get triggered. For example... Cxbx-Reloaded doesn't support KPCR across all threads and will produce failure.

Reply:
I'm marking them as invalid due to insufficient reading the whole tests.

Reply:
My other comment was meant for this line, where there IS an unexplained `0xFF` value being tested against (and another one 3 lines down).

Besides there not being an expansion on why exactly the value `0xFF` is present/needed here, the first part of the check could be removed (and similarly 3 lines down), because if the second check succeeds, the first part is always `true` too and thus unnecessary.

Reply:
See comment below review remark of its intentional design.